### PR TITLE
NFT UI

### PR DESCRIPTION
**Pull request by CodeWizard224**
### Conflicts:
	app/src/main/java/io/novafoundation/nova/app/root/navigation/nft/NftNavigator.kt
	core-db/src/main/java/io/novafoundation/nova/core_db/dao/NftDao.kt
	feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/receive/ReceiveFragment.kt
	feature-nft-api/src/main/java/io/novafoundation/nova/feature_nft_api/data/repository/NftRepository.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/repository/NftRepositoryImpl.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/source/NftProvider.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/source/providers/rmrkV1/RmrkV1NftProvider.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/source/providers/rmrkV2/RmrkV2NftProvider.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/data/source/providers/uniques/UniquesNftProvider.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/list/NftListInteractor.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/send/NftSendInteractor.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/presentation/nft/details/NftDetailsFragment.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/presentation/nft/list/NftGridAdapter.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/presentation/nft/send/confirm/ConfirmNftSendFragment.kt
	feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/presentation/nft/send/confirm/ConfirmNftSendViewModel.kt

### CODE REVIEW

File: feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/common/Mappers.kt [hunk 1]
```diff
@@ -7,8 +7,16 @@ fun mapNftDetailsToListItem(nftDetails: NftDetails): SendNftListItem {
...
         name = nftDetails.name,
-        collectionName = nftDetails.collection?.name ?: nftDetails.collection?.id.orEmpty(),
+        collectionName = mapNftCollectionForUi(nftDetails.collection?.name, nftDetails.collection?.id),
         media = nftDetails.media,
         chain = nftDetails.chain
     )
 }
+
+fun mapNftCollectionForUi(collectionName: String?, collectionId: String?): String {
+    return collectionName ?: "NFT Collection ID: $collectionId"
```
Comment (palmut) [inline feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/common/Mappers.kt:17]: хочется тут из ресурсов подтянуть локализованную строку, но это пока только ощущения

File: feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/list/PricedNft.kt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package io.novafoundation.nova.feature_nft_impl.domain.nft.list
 
 import io.novafoundation.nova.feature_nft_api.data.model.Nft
+import io.novafoundation.nova.feature_nft_api.data.model.NftDetails
```
Comment (palmut) [inline feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/list/PricedNft.kt]: нужен импорт этот?

**Review by palmut — APPROVED** [review]

File: feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/common/Mappers.kt (hunk 1 above)
Comment (CodeWizard224) [reply feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/common/Mappers.kt:17]: Так как mapping происходит на уровне Interactor и нет контекста при текущей архитектуре приложения сложно использовать ресурсы. Предложение хорошее, конечно

File: feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/list/PricedNft.kt (hunk 2 above)
Comment (CodeWizard224) [reply feature-nft-impl/src/main/java/io/novafoundation/nova/feature_nft_impl/domain/nft/list/PricedNft.kt]: Пофиксил

**Pull request merged**